tchannel/peer_heap.py

```diff
@@ -80,3 +80,8 @@ def peek_peer(self):
             return None
 
         return self.peers[0]
+
+    def remove_peer(self, peer):
+        """Remove the peer from the heap."""
+        if peer.index < self.size():
```

Comment:
Is it worth verifying that the peer being removed is the same peer? Are we guaranteed that the peer.index will never be out of date?